### ISSUE TITLE
Accessibility: GridField should use `aria-sort` attribute on sorted column header

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-framework#11860]
The `aria-sort` attribute should be set to "ascending" or "descending" on the `th` element that contains the header for the currently sorted column (if any).

## Notes

- Sortable headers for asset admin got this added in https://github.com/silverstripe/silverstripe-asset-admin/issues/1611
- If there are any other sortable headers, they should get the same treatment.

## References

- https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-sort

## Acceptance Criteria

- [ ] Relevant `th` element uses `aria-sort` attribute
- [ ] `aria-sort` attribute gets updated when sort is updated
- [ ] If no column is currently sorted, no element uses the `aria-sort` attribute.

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/11896
- https://github.com/silverstripe/silverstripe-asset-admin/pull/1631
- https://github.com/silverstripe/developer-docs/pull/837

**Comment by GuySartorelli** [comment]
PRs merged
